### PR TITLE
CortexStep available in "next"

### PR DESCRIPTION
**Pull request by tobowers**
This takes our closed source works and makes it available here. The major change here is using functions and typed outputs (but still allowing regular text/streaming). More details available in the new repo. After this merge, we can work on new examples.

### CODE REVIEW

File: core/src/next/README.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+Since this repository is private, you'll need to authorize npm to pull from github packages for the @opensouls scope.
+To authorize npm, you need to login with the following command:
+
+```bash
+npm login --scope=@opensouls --auth-type=legacy --registry=https://npm.pkg.github.com`
+```
+
+When prompted, your username is your github username and your password is your personal access token (classic) with with the following scopes: `read:packages`
+You can create a personal access token here: https://github.com/settings/tokens
```
Comment (kafischer) [inline core/src/next/README.md:87]:
```suggestion
```

File: core/src/next/cognitiveFunctions.ts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            content: `${step.entityName} ${action} ${response[action]}`
+          }],
+        }
+      }
+    };
+  }
+}
+
+export const externalDialog = singleResponse
+export const internalMonologue = singleResponse
```
Comment (kafischer) [inline core/src/next/cognitiveFunctions.ts:30]: i'm not sure if we should include these ones per say since we know they have major flaws

Comment (tobowers) [reply core/src/next/cognitiveFunctions.ts:30]: yeah good call. I have a branch trying to fix that already in the closed repo, so maybe I can bring that work here.

**Comment by tobowers** [comment]
@kafischer - ok I moved the internal/external to the *tests* so that we can continue to iterate on them.

File: core/tests/next/CortexStep.spec.ts [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
+        name: `create_and_save_${action}`.replace(/\s/g, "_"),
+        description,
+        parameters: params,
+        command: html`
+          Analyze the chat history and answer from the first person perspective and voice of ${entityName}.
+          
+          Answer in the first person perspective and voice of ${entityName} the completion to: ${entityName} ${action}
+
+          For example:
+          ${entityName} ${action} I will win this game!
```
Comment (kafischer) [inline core/tests/next/CortexStep.spec.ts:30]: i suspect we want some quotes here for clarity?

Comment (tobowers) [reply core/tests/next/CortexStep.spec.ts]: I've experimented with nothing, with colons, and with quotes. I think colons tend to work the best (in other scenarios). I adjusted to add it here. Didn't make a visible difference to the test results, but maybe would in longer contexts.

Comment (kafischer) [reply core/tests/next/CortexStep.spec.ts]: i think it's also partially nice even just for a human :)

**Comment by tobowers** [comment]
so we good to merge?

**Pull request merged**